### PR TITLE
Add tests for KarmaController

### LINKED ISSUE #65: Add tests for KarmaController
This we not be perfect but wanted to dive into adding tests in order to begin the process on adding CI to the project in order to catch breaking changes sooner

### PR DESCRIPTION
**Pull request by jwilson-slalom**
Resolves #65 

~~This is not perfect. We are missing testing logic that happens when slash commands are used. It just needs some more abstraction to capture the POST call that is sent to the `response_url` and I felt like A. that was complicated and B. this PR was already going to be very big.~~

Edit: #69 will now handle this edge case and will therefore be tested

### CODE REVIEW

File: Tests/AppTests/Controllers/KarmaControllerTests.swift [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        let request = validatedSlackRequest(using: app)
+        do {
+            let status = try controller.command(request, content: command)
+            XCTAssertEqual(status, .ok)
+        } catch {
+            XCTFail(error.localizedDescription)
+        }
+    }
+
+    func testThatItHandlesWhenUserTriesToAdjustThemselves() {
```
Comment (johnwelchatslalomcom) [inline Tests/AppTests/Controllers/KarmaControllerTests.swift]: I'd say a better description of the requirement is:

```swift
func testThatItDoesNotAllowAUserToAdjustKarmaForSelf()
```

File: Sources/App/Controllers/Karma/KarmaController+Slack.swift [hunk 2]
```diff
@@ -57,7 +57,7 @@ extension KarmaController {
                 }
             }.catch {
                 self.log.error("Failed to respond to Slack slash command \($0)")
-        }
+            }
```
Comment (allenhumphreys) [inline Sources/App/Controllers/Karma/KarmaController+Slack.swift]: Nice

File: Tests/AppTests/Controllers/KarmaControllerTests.swift [hunk 3]
```diff
@@ -0,0 +1,281 @@
+//
+//  KarmaControllerTests.swift
+//  AppTests
+//
+//  Created by Jacob Wilson on 3/12/19.
+//
+
+@testable import App
+import Vapor
+import struct SlackKit.User
```
Comment (allenhumphreys) [inline Tests/AppTests/Controllers/KarmaControllerTests.swift]: Why are we importing SlackKit stuff into the tests?
